app/controllers/match_user_by_email_controller.rb

```diff
@@ -0,0 +1,34 @@
+class MatchUserByEmailController < ApplicationController
+  before_action :fetch_session_if_present
+
+  def show
+    email = params.fetch(:email).downcase
+
+    if @govuk_account_session&.user&.email == email
+      render json: { match: true }
+      return
+    end
+
+    if OidcUser.where(email: email).exists?
+      render json: { match: false }
+    else
+      head :not_found
```

Comment:
So a logged in user could use this endpoint to inferr if there was another account by email, right?
We're not limiting checks to just the account that is making the query?

I guess you could do the same with our login form, and at least this way we'd have a logged in user making this request.

Should we consider rate limiting this at all?

Reply:
email-alert-frontend has a rate limiter: https://github.com/alphagov/email-alert-frontend/blob/main/app/services/verify_subscriber_email_service.rb - we could re-use that.

And yes, this intentionally doesn't require a logged in user, because for the Taxon / Topic Updates journey we *don't* have a logged-in user at the point where we look up an email address, and I wanted this endpoint to work for both (rather than have two very-similar-but-slightly-different endpoints)